poweriq-worker/src/main/java/com/vmware/flowgate/poweriqworker/jobs/PowerIQService.java

```diff
@@ -1185,4 +1198,303 @@ public void syncPDUID(FacilitySoftwareConfig powerIQ) {
+            String maxVoltageValue = null;
+            if(rangeFlagIndex != -1) {
+               minVoltageValue = voltageValue.substring(0, rangeFlagIndex);
+               maxVoltageValue = voltageValue.substring(rangeFlagIndex + 1);
+               pduRateInfo.put(FlowgateConstant.PDU_MIN_RATE_VOLTS, minVoltageValue);
+               pduRateInfo.put(FlowgateConstant.PDU_MAX_RATE_VOLTS, maxVoltageValue);
+            }else {
+               //There is a another format of voltage value like 200V
+               minVoltageValue = voltageValue;
+               maxVoltageValue = voltageValue;
+               pduRateInfo.put(FlowgateConstant.PDU_MIN_RATE_VOLTS, minVoltageValue);
+               pduRateInfo.put(FlowgateConstant.PDU_MAX_RATE_VOLTS, maxVoltageValue);
+            }
+         }else {
+            logger.error(String.format("Invalid value for rate voltage : %s",rateVolts));
+         }
+      }
+      return pduRateInfo;
+   }
+
+   private List<Asset> getAllPduAssetsFromPowerIQ(FacilitySoftwareConfig powerIQ) {
+      int limit = 100;
+      int offset = 0;
+      PowerIQAPIClient client = createClient(powerIQ);
+      List<Pdu> pdus = null;
+      List<Asset> pduAssets = new ArrayList<>();
+      Map<Long, Rack> racksMap = getRacksMap(client);
+      Map<Long, Row> rowsMap = getRowsMap(client);
+      Map<Long, Aisle> aislesMap = getAislesMap(client);
+      Map<Long, Room> roomsMap = getRoomsMap(client);
+      Map<Long, Floor> floorsMap = getFloorsMap(client);
+      Map<Long, DataCenter> dataCentersMap = getDataCentersMap(client);
+
+      while ((pdus = client.getPdus(limit, offset)) != null) {
+         if (pdus.isEmpty()) {
+            break;
+         }
+         for (Pdu pdu : pdus) {
+            Asset asset = new Asset();
+            asset = fillLocation(pdu.getParent(), racksMap, rowsMap, aislesMap, roomsMap, floorsMap,
+                  dataCentersMap);
+            asset.setAssetName(pdu.getName());
+            asset.setSerialnumber(pdu.getSerialNumber());
+            asset.setAssetSource(powerIQ.getId());
+            asset.setCategory(AssetCategory.PDU);
+            asset.setCreated(System.currentTimeMillis());
+            List<Outlet> outlets = client.getOutlets(pdu.getId());
+            List<Inlet> inlets = client.getInlets(pdu.getId());
+            ObjectMapper mapper = new ObjectMapper();
+            String outletString = null;
+            String inletString = null;
+            try {
+               outletString = generatePduOutletString(outlets);
+               inletString = generatePduInletString(inlets);
+            } catch (JsonProcessingException e) {
+               logger.info(String.format("Sync pdu metadata error",e.getCause()));
+            }
+            Map<String,String> pduInfo = generatePduRateInfoMap(pdu);
+            if(outletString != null) {
+               pduInfo.put(FlowgateConstant.PDU_OUTLETS_FROM_POWERIQ, outletString);
+            }
+            if(inletString != null) {
```

Comment:
I think you can simply pass the asset into the generatedPDURateInfo method. then you don't need to double check whether outletString is empty once again.

Same for the outlet.  This will make it more clear and remove a lot of obsolete code.  

Let's write unit test case for your code.